frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx

```diff
@@ -0,0 +1,181 @@
+  initialSizeUnit = 'Gi',
+  initialVolumeMode = 'Filesystem',
+}) => {
+  const [accessModeHelp, setAccessModeHelp] = React.useState('Permissions to the mounted drive.');
+  const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const [allowedAccessModes, setAllowedAccessModes] = React.useState(initialAccessModes);
+  const [volumeMode, setVolumeMode] = React.useState(initialVolumeMode);
+  const [accessMode, setAccessMode] = React.useState('ReadWriteOnce');
+  const [requestSizeValue, setRequestSizeValue] = React.useState(initialSizeValue);
+  const [requestSizeUnit, setRequestSizeUnit] = React.useState(initialSizeUnit);
+  const [storageProvisioner, setStorageProvisioner] = React.useState('');
+  const [storageClass, setStorageClass] = React.useState(initialSizeUnit);
+
+  const { t } = useTranslation();
+
+  const handleAccessMode: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    setAccessMode(event.currentTarget.value);
+  };
+
+  const handleVolumeMode: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    setVolumeMode(event.currentTarget.value);
+  };
+
+  const handleStorageClass = (updatedStorageClass) => {
+    const provisioner: string = updatedStorageClass?.provisioner || '';
+    // if the provisioner is unknown or no storage class selected, user should be able to set any access mode
+    const modes = provisionerAccessModeMapping[provisioner]
+      ? provisionerAccessModeMapping[provisioner]
+      : getAccessModeForProvisioner(provisioner);
+    // setting message to display for various modes when a storage class of a know provisioner is selected
+    const displayMessage =
+      provisionerAccessModeMapping[provisioner] || isCephProvisioner(provisioner)
+        ? 'Access mode is set by storage class and cannot be changed'
+        : 'Permissions to the mounted drive';
+    setAccessMode('ReadWriteOnce');
+    setAccessModeHelp(displayMessage);
+    // setting accessMode to default with the change to Storage Class selection
+    setAllowedAccessModes(modes);
+    setStorageClass(updatedStorageClass?.metadata?.name);
+    setStorageProvisioner(provisioner);
+    if (storageProvisioner.includes(cephStorageProvisioners[1])) {
+      setVolumeMode('Filesystem');
+    }
+  };
+
+  React.useEffect(() => {
+    const volumeClaimTemplate = {
+      spec: {
+        accessModes: [accessMode],
+        storageClassName: storageClass,
+        volumeMode,
+        resources: {
+          requests: {
+            storage: `${requestSizeValue}${requestSizeUnit}`,
+          },
+        },
+      },
+    };
+    setFieldTouched(name);
+    setFieldValue(name, volumeClaimTemplate);
+  }, [
+    volumeMode,
+    accessMode,
+    requestSizeValue,
+    requestSizeUnit,
+    storageClass,
+    name,
+    setFieldTouched,
+    setFieldValue,
+  ]);
+
+  return (
+    <>
+      <p className="help-block">
+        {t('pipelines-plugin~This will create a PersistentVolumeClaim with a size of 1 GiB')}
```

Comment:
Maybe we use the actual value here... That way when they change it, the wording doesn't say the incorrect thing.

Reply:
yes, that seems to be the better way. Updated accordingly